SimG4Core/Notification/interface/SimTrackManager.h

```diff
@@ -101,6 +101,27 @@ class SimTrackManager {
     }
     return flag;
   }
+  TrackWithHistory* getTrackByID(unsigned int trackID) const {
+    bool trackFound = false;
+    TrackWithHistory* track;
+    if (m_trksForThisEvent == nullptr) {
```

Comment:
there is no reason for exception

Reply:
I added an optional parameter `bool strict=false`. If `strict=true`, the exception is still raised. The reason is that this particular exception has been immensely helpful for debugging finecalo and I'd like to keep the option to raise it.

Reply:
Is the `strict` option supposed to be used for this exception, or just the one further down?

Reply:
The one further down. Basically it's meant to raise when you try to get an invalid ID.